engines/dfc_provider/app/controllers/dfc_provider/api/enterprises_controller.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+# Controller used to provide the CatalogItem API for the DFC application
+module DfcProvider
+  module Api
+    class EnterprisesController < DfcProvider::Api::BaseController
+      def show
+        render json: current_enterprise, serializer: DfcProvider::EnterpriseSerializer
+      end
+
+      private
+
+      def current_enterprise
+        @current_enterprise ||=
+          if params[:id] == 'default'
+            current_user.enterprises.first!
+          else
+            current_user.enterprises.find(params[:id])
+          end
```

Comment:
you can avoid this copy pasted code here.

on the base_controller have:
```ruby
def current_enterprise
  @current_enterprise ||=
          if params[enterprise_id_param_name] == 'default'
            current_user.enterprises.first!
          else
            current_user.enterprises.find(params[enterprise_id_param_name])
          end
end

protected

def enterprise_id_param_name
    :enterprise_id
end
```

and then here on the enterprise controller delete current_enterprise and just have:
```ruby
protected

def enterprise_id_param_name
    :id
end
```

Reply:
Yes, I could DRY this, I was not sure about it as it hides a bit the logic, but it is not that complex for sure.